core/src/actions/destinations.ts

```diff
@@ -215,19 +221,15 @@ export class DestinationConnectionOptions extends AuthenticatedAction {
     this.permission = { topic: "destination", mode: "read" };
     this.inputs = {
       id: { required: true },
-      options: { required: false },
+      options: { required: false, formatter: APIData.ensureObject },
     };
   }
 
   async runWithinTransaction({ params }) {
     const destination = await Destination.findById(params.id);
-
-    const options =
-      typeof params.options === "string"
-        ? JSON.parse(params.options)
-        : params.options;
```

Comment:
Previously, this type of JSON parsing logic was inconsistently scattered around the codebase.  Now our formatters do the work for us!